### PR TITLE
getAllUpcomingEvents and getAllPastEvents

### LINKED ISSUE #117: getAllUpcomingEvents
- Endpoint URI: */event/upcoming*
- Http Method: *GET*
- API Logic: 
    - Find all events with Open and Postpone status and the start date is later from Instant.now().

- Progress
    - [x] development
    - [x] integration Test

### LINKED ISSUE #118: getAllPastEvents
- Endpoint URI: */event/past*
- Http Method: *GET*
- API Logic: 
    - Find all events with Open and Postpone status and the start date is before Instant.now().

- Progress
    - [x] development
    - [x] integration Test

### PR DESCRIPTION
**Pull request by junzai97**
Resolved #117 
Resolved #118

### CODE REVIEW

File: src/main/java/com/thirdcc/webapp/service/impl/EventServiceImpl.java [hunk 1]
```diff
@@ -76,6 +76,45 @@ public EventDTO save(EventDTO eventDTO) {
...
+     * Get all the past events.
+     *
+     * @return the list of entity.
+     */
+    @Override
+    public Page<EventDTO> findAllPastEvents(Pageable pageable) {
+        log.debug("Request to get all Past Events");
+
+        Set<EventStatus> eventStatuses = new HashSet<EventStatus>() {{
+            add(EventStatus.CANCELLED);
```
Comment (HowardYaw) [inline src/main/java/com/thirdcc/webapp/service/impl/EventServiceImpl.java:115]: If CANCELLED why still need to show in Previous Event?

File: src/test/java/com/thirdcc/webapp/web/rest/EventResourceIT.java [hunk 2]
```diff
@@ -242,6 +240,54 @@ public void getAllEvents() throws Exception {
...
+    @Test
+    @Transactional
+    public void getAllUpcomingEvents() throws Exception {
+        // Initialize the database
+        Event savedEvent = createEntity();
+        savedEvent.setStartDate(Instant.now().plus(20, ChronoUnit.DAYS));
+        eventRepository.saveAndFlush(savedEvent);
+
+        // Get all the eventList
+        restEventMockMvc.perform(get("/api/events?sort=id,desc"))
```
Comment (HowardYaw) [inline src/test/java/com/thirdcc/webapp/web/rest/EventResourceIT.java]: this API correct?

File: src/test/java/com/thirdcc/webapp/web/rest/EventResourceIT.java [hunk 3]
```diff
@@ -242,6 +240,54 @@ public void getAllEvents() throws Exception {
...
+    @Test
+    @Transactional
+    public void getAllPastEvents() throws Exception {
+        // Initialize the database
+        Event savedEvent = createEntity();
+        savedEvent.setStartDate(Instant.now().minus(20, ChronoUnit.DAYS));
+        eventRepository.saveAndFlush(savedEvent);
+
+        // Get all the eventList
+        restEventMockMvc.perform(get("/api/events?sort=id,desc"))
```
Comment (HowardYaw) [inline src/test/java/com/thirdcc/webapp/web/rest/EventResourceIT.java]: this API correct?

**Review by HowardYaw — CHANGES_REQUESTED** [review]

File: src/main/java/com/thirdcc/webapp/service/impl/EventServiceImpl.java (hunk 1 above)
Comment (junzai97) [reply src/main/java/com/thirdcc/webapp/service/impl/EventServiceImpl.java:115]: So means if the event is cancelled, the admin can't visit the event again?
My concern is that when an admin accidently changed the status to cancelled, the admin should can look for the event and change back the status.

File: src/test/java/com/thirdcc/webapp/web/rest/EventResourceIT.java (hunk 2 above)
Comment (junzai97) [reply src/test/java/com/thirdcc/webapp/web/rest/EventResourceIT.java]: Fixed in 0776434d822e3f44812360d825e6a04ca8933805

File: src/test/java/com/thirdcc/webapp/web/rest/EventResourceIT.java (hunk 3 above)
Comment (junzai97) [reply src/test/java/com/thirdcc/webapp/web/rest/EventResourceIT.java]: Fixed in 0776434d822e3f44812360d825e6a04ca8933805

File: src/main/java/com/thirdcc/webapp/service/impl/EventServiceImpl.java (hunk 1 above)
Comment (HowardYaw) [reply src/main/java/com/thirdcc/webapp/service/impl/EventServiceImpl.java:115]: we can have a discussion on this on Friday
so far is ok I think

**Review by HowardYaw — DISMISSED** [review]

File: src/test/java/com/thirdcc/webapp/web/rest/EventAttendeeResourceIT.java [hunk 4]
```diff
@@ -532,29 +532,19 @@ public void getEventAttendeeByEventIdAndUserId_withNonExistingUser_ShouldReturnO
 
         //Save events and attendee
         restEventAttendeeMockMvc.perform(get("/api/event-attendees/event/{eventId}/user/{userId}", savedEvent.getId(), Long.MAX_VALUE ))
-            .andExpect(status().isOk())
-            .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
-            .andExpect(jsonPath("$.id").value(nullValue()))
-            .andExpect(jsonPath("$.userId").value(nullValue()))
-            .andExpect(jsonPath("$.eventId").value(nullValue()))
-            .andExpect(jsonPath("$.provideTransport").value(nullValue()));
```
Comment (HowardYaw) [inline src/test/java/com/thirdcc/webapp/web/rest/EventAttendeeResourceIT.java]: Why not keep this?

Comment (junzai97) [reply src/test/java/com/thirdcc/webapp/web/rest/EventAttendeeResourceIT.java]: Resolved in c9e67c7c1113b73f838af40f9d8987ca995becae

**Review by HowardYaw — APPROVED** [review]

**Review by LUXIANZE — APPROVED** [review]

**Pull request merged**